homeassistant/components/myuplink/sensor.py

```diff
@@ -67,14 +67,22 @@ def get_description(device_point: DevicePoint) -> SensorEntityDescription | None
     """Get description for a device point.
 
     Priorities:
-    1. Category specific parameter_id e.g "NIBEF F730 CU 3x400V-49994"
+    1. Category specific prefix e.g "NIBEF-49994"
     2. Global parameter_id e.g. "49994"
     3. Global parameter_unit e.g. "°C"
     4. Default to None
     """
-    description = DEVICE_POINT_DESCRIPTIONS.get(
-        f"{device_point.category}-{device_point.parameter_id}"
-    )
+    model_prefix = ""
+    for prefix in ("NIBEF",):
+        if device_point.category.startswith(prefix):
```

Comment:
Could we calculate the prefix by only using the `device_point.category`? Eg by looking for a separator in the string or taking a certain number of characters from the beginning etc?

I'd like to avoid having a tuple of known prefixes to match against and needing to maintain that tuple.

Reply:
This is a difficult one...
There is no simple way to categorize the models. The product,name appears on different levels in the object hierarchy and the device_point.category is not an exact match to the product.name.
My interpretation of the data model in the API:

1. An account can containg many systems.
2. A system is a heat-pump appliance.
3. A system can contain one or more devices
4. A device contains many device-point.
5. There is no official documentation from the manufacturer on the actual data that each model reports.

Example from my F730:
One system with one device with some 45 device points.
product.name in system object: "F730 CU 3x400V"
a single device with product.name: "F730 CU 3x400V"
category in all device_points: "NIBEF F730 CU 3x400V"

Example from @pajzo:
product.name: unknown to me
category in device_points: Mostly "NIBE VVM S320 E EM 3x400V DK" but mixed with "Varmepumpe 1"

I am out of ideas (for a short moment - I hope). I am a bit sorry that the search for a perfect solution is blocking other areas of improvements. It is possible to create generic binary_sensors, generic switches and other suitable entities. It is when it comes to fine-tuning device_classes, icons, display precision etc that we are stuck.

The current solution will work for Fxxx -series appliances. All other models will show up with generic sensors as in the original release.
I suggest that we merge this version. We can then revisit the challenge of classifying appliance models when we have collected more diagnostic data.

Reply:
I agree with you @astrandb, i think it's fine to map based on parameter_id, and fallback on paramter_unit. Then we over time can improve by adding more and more parameter_id mappings.

Reply:
We now have confirmation thay S series have a different numbering scheme which may overlap. So we must discern them.

We also now see that S series prefix by "NIBE" and F series by "NIBEF". That is enough.

Just use partition on the category and grab forst words. Use a double level dict. For lookup.

Also avoid putting unit lookups and paramater id lookups in same dict.

prefix, _, _ = partition(category, " ")
description = database.get(prefix, {}).get(parameter_id)
if not description:
  description = unitdb.get(unit)

Reply:
Thank you @elupus for you constructive ideas. They are implemented now.